src/game_api/script/usertypes/level_lua.cpp

```diff
@@ -189,6 +189,752 @@ LevelGenRoomData& PreHandleRoomTilesContext::get_mutable_room_data()
+        }
+        return std::nullopt;
+    }
+
+    template <class Ret, class... Args>
+    std::optional<Ret> run_post_func(THEME_OVERRIDE index, Args&&... args)
+    {
+        if (overrides.find(index) != overrides.end() && get_post_func_enabled(index))
+        {
+            return backend->handle_function_with_return<Ret>(overrides[index]->post.value(), std::forward<Args>(args)...);
+        }
+        return std::nullopt;
+    }
+
+    CustomTheme(uint8_t theme_id_, uint8_t base_theme_, bool defaults)
+    {
+        level_file = "";
+        theme = theme_id_;
+        base_theme = base_theme_ - 1;
+        backend = LuaBackend::get_calling_backend();
+        if (defaults)
+        {
+            overrides[THEME_OVERRIDE::INIT_FLAGS] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_LEVEL] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_PLAYERS] = new ThemeOverride{true, 0};
+            overrides[THEME_OVERRIDE::SPAWN_TRANSITION] = new ThemeOverride{true, 0};
+            overrides[THEME_OVERRIDE::POST_TRANSITION] = new ThemeOverride{true, 0};
+            overrides[THEME_OVERRIDE::SPAWN_PROCEDURAL] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_BACKGROUND] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_BORDER] = new ThemeOverride{true, 0};
+            overrides[THEME_OVERRIDE::ENT_BORDER] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::TEXTURE_DYNAMIC] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::PLAYER_DAMAGE] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::GRAVITY] = new ThemeOverride{true, UINT8_MAX};
+            /*
+            overrides[THEME_OVERRIDE::INIT_LEVEL] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::INIT_FLAGS] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_EFFECTS] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::TRANSITION_MODIFIER] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::UNKNOWN_V32] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::UNKNOWN_V38] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::UNKNOWN_V47] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_DECORATION] = new ThemeOverride{true, UINT8_MAX};
+            overrides[THEME_OVERRIDE::SPAWN_DECORATION2] = new ThemeOverride{true, UINT8_MAX};
+            */
+        }
+    }
+    CustomTheme(uint8_t theme_id_, uint8_t base_theme_)
+        : CustomTheme(theme_id_, base_theme_, true)
+    {
+    }
+    CustomTheme()
+        : CustomTheme(100, 1, true)
+    {
+    }
+    ~CustomTheme()
+    {
+    }
+    bool get_unknown1()
+    {
+        return unknown1;
+    }
+    void init_flags()
```

Comment:
Maybe do some sort of macro or template here to avoid the amount of duplicate code for each function?

Reply:
Yeah especially the void ones with no special cases could use a template.